apps/organisasjon-orgnummer-service/src/main/java/no/nav/registre/orgnrservice/consumer/OrganisasjonApiConsumer.java

```diff
@@ -30,49 +28,53 @@
     private final String clientId;
     private final AccessTokenService accessTokenService;
     private final ExecutorService executorService;
+    private final MiljoerConsumer miljoerConsumer;
 
-    OrganisasjonApiConsumer (@Value("${consumers.organisasjon-api.url}") String url,
-                             @Value("${consumers.organisasjon-api.client_id}") String clientId,
-                             @Value("${consumers.organisasjon-api.threads}") Integer threads,
-                             AccessTokenService accessTokenService) {
+    OrganisasjonApiConsumer(@Value("${consumers.organisasjon-api.url}") String url,
+                            @Value("${consumers.organisasjon-api.client_id}") String clientId,
+                            @Value("${consumers.organisasjon-api.threads}") Integer threads,
+                            AccessTokenService accessTokenService, MiljoerConsumer miljoerConsumer) {
         this.clientId = clientId;
         this.accessTokenService = accessTokenService;
         this.executorService = Executors.newFixedThreadPool(threads);
+        this.miljoerConsumer = miljoerConsumer;
         this.webClient = WebClient.builder()
                 .baseUrl(url)
                 .build();
     }
 
-    CompletableFuture<OrganisasjonDTO> getOrgnrFraMiljoeThreads (String orgnummer, String miljoe, String token) {
+    CompletableFuture<OrganisasjonDTO> getOrgnrFraMiljoeThreads(String orgnummer, String miljoe, String token) {
         return CompletableFuture.supplyAsync(
                 () -> new GetOrganisasjonCommand(webClient, token, orgnummer, miljoe).call(),
                 executorService
         );
     }
 
     public boolean finnesOrgnrIEreg(String orgnummer) {
-        List<String> miljoer = Arrays.asList("q0", "q1", "q2");
+
+        List<String> miljoer = miljoerConsumer.hentMiljoer().getEnvironments();
 
         var futures = miljoer.stream()
-                .map( enkeltmiljoe -> getOrgnrFraMiljoeThreads(orgnummer, enkeltmiljoe, getToken()))
+                .filter(miljoe -> !miljoe.equals("qx") && !miljoe.equals("u5"))
```

Comment:
Kan eventuelt lage et array med beskrivende navn for qx og u5 🤷

Reply:
Supert 😄